activerecord/lib/active_record/fixture_set/table_row.rb

```diff
@@ -117,14 +117,28 @@ def interpolate_label
         end
 
         def generate_primary_key
-          # generate a primary key if necessary
-          Array(model_metadata.primary_key_name).each do |pk|
-            next if !model_metadata.has_column?(pk) || @row.include?(pk)
+          pk = model_metadata.primary_key_name
 
+          unless column_defined?(pk)
             @row[pk] = ActiveRecord::FixtureSet.identify(@label, model_metadata.column_type(pk))
           end
         end
 
+        def generate_composite_primary_key
+          id = ActiveRecord::FixtureSet.identify(@label)
+          model_metadata.primary_key_name.each_with_index do |column, index|
+            next if column_defined?(column)
+            raise "Automatic key generation assumes columns of type Integer." unless model_metadata.column_type(column) == :integer
```

Comment:
If someone ever runs into this limitation and actually needs non-integer parts of the primary key to be autogenerated then we could consider the following alternative:
```ruby
    pk_columns = [:column1, :column2]
    label = "my_order"
    row = { name: "First order" }

    pk_columns.map.with_index do |pk, i|
      column_type = :integer # simplified for the demo but should be actual column type fetch

      # Design-wise this check is unnecessary but we want to generate the same
      # primary key value for existing fixtures
      label = i == 0 ? label : "#{label}_pk#{i}"
      row[pk] = ActiveRecord::FixtureSet.identify(pk, column_type)
    end
```

This approach should allow us to unify generation for all column types and primary key sizes

Reply:
I explored this option, and while I think adding a hidden suffix for each component of the label is a good idea, I don't think it would bring support for non-integer types.

It's important to note: [`ActiveRecord::FixtureSet.identify` only returns integers.](https://github.com/rails/rails/blob/2cef3ac45bd85e1642b1270ab4fa31629044c962/activerecord/lib/active_record/fixtures.rb#L580-L586)

The column type passed to the method is used to either return a UUID or an Integer and changes the hashing method based on this. Therefore, other types, such as Strings, Booleans, Floats would not be considered.

Reply:
Ah, right, I forgot that that we strictly check for `column_type == :uuid` 
Though conceptually we could use an UUID string as a random value for a string-based column type
But at this point it's hard to come up with a use-case where an application would prefer a random UUID as a value for a string column instead of an explicitly value with some meaning 
But as long as we don't generate anything apart `uuid` and `integer` - current solution is totally fine 👍